src/file-upload/file-option-thumbnail.tsx

```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useState } from 'react';
+import styles from './styles.css.js';
+
+interface FileOptionThumbnailProps {
+  file: File;
+}
+
+export function FileOptionThumbnail({ file }: FileOptionThumbnailProps) {
+  const [imageSrc, setImageSrc] = useState('');
+
+  useEffect(() => {
+    if (URL.createObjectURL) {
+      const src = URL.createObjectURL(file);
+      setImageSrc(src);
+
+      return () => {
+        URL.revokeObjectURL(src);
+      };
+    }
+  }, [file]);
+
+  const onImageLoad = () => {
+    URL.revokeObjectURL(imageSrc);
```

Comment:
I used this example as a reference: https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications#example_using_object_urls_to_display_images

Reply:
> Set up the image's load event handler to release the object URL since it's no longer needed once the image has been loaded. This is done by calling the [URL.revokeObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL) method and passing in the object URL string as specified by img.src.

This breaks "open in a new tab" and "save image" features. Not sure why would anyone save their own local file one more time, but also unclear what is the harm of preserving the value.

I think they do it because they have no "unmount" lifecycle, so they want to prevent memory leaks. In our case it is not needed

Reply:
Agreed - will only keep the revoke in the effect cleanup.